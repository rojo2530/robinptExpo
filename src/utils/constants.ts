export const API_URL = 'https://stores.robinrover.io/api';
export const API_RESULTS_PER_PAGE = 10;
export const TAB_BAR_ICON_SIZE = 24;

export enum TABS_ROUTE_NAME {
  SHOPS = 'ShopsList',
  FAVORITES = 'ShopsFavoriteList'
}

export const TAB_SHOPS_LIST = 'ShopsList';

export const Colors = {      
  // MAIN COLORS
  main: {
    PRIMARY : '#1B9BD7',
    SECONDARY : '#f4511e',  
  },
  // GRAYSCALE
  grayscale:   {
    BLACK : '#000000',
    GRAY_LIGHT : '#F6F6F6',
    GRAY_MEDIUM : '#cacaca',
    GRAY_DARK : '#8a8a8a',
    GRAY_TYPO : '#888686',
    WHITE : '#ffffff',
  },
  actions:   {
    SUCCESS : '#47A665',
    WARNING : '#ffae00',
    ALERT : '#D24858',
    ALERT_TRANSPARENT : '#FFECEC',
    WARNING_TRANSPARENT: '#ffd780',
    SUCCESS_TRANSPARENT: '#94d1a7',
    YELLOW : '#F9C962',
  },
};