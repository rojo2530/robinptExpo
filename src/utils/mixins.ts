import { PixelRatio } from 'react-native';

export const scaleFont = (size: number) => size * PixelRatio.getFontScale();