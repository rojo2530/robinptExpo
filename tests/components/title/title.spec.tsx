import * as React from "react"
import * as Testrenderer from "react-test-renderer"
import Title from "../../../src/components/title/title";
import { Colors } from "../../../src/utils/constants";


describe("<Title />", () => {
  it("has 0 child", () => {
    const tree: any = Testrenderer.create(<Title />).toJSON();
    expect(tree.children).toBe(null);
  });

  it("Match SnapShot", () => {
    const title: any = Testrenderer.create(<Title />);
    expect(title.toJSON()).toMatchSnapshot();
  });

  it("has default props", () => {
    const title: any = Testrenderer.create(<Title />);
    const instance = title.root;
    const findAllByNumberOfLines = (instance: any) => instance.findAll((el: any) => 
      el.props.numberOfLines === 3 &&
      el.type === 'Text'
    );

    const text = instance.findByType("Text");
    expect(text.props.style).toHaveProperty("fontSize", 22);
    expect(text.props.style).toHaveProperty("color", Colors.grayscale.BLACK);
    expect(findAllByNumberOfLines(instance).length).toEqual(1);
  });
});