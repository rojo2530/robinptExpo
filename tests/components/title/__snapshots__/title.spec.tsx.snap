// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Match SnapShot 1`] = `
<Text
  numberOfLines={3}
  style={
    Object {
      "color": "#000000",
      "fontSize": 22,
    }
  }
/>
`;
